code/modules/photography/camera/silicon_camera.dm

```diff
@@ -89,3 +89,13 @@
 	C.toner -= printcost  //All fun allowed.
 	visible_message(span_notice("[C.name] spits out a photograph from a narrow slot on its chassis."))
 	to_chat(usr, span_notice("You print a photograph."))
+
+/obj/item/camera/siliconcam/proc/paiprint(mob/user)
+	var/mob/living/silicon/pai/paimob = loc
+	var/datum/picture/selection = selectpicture(user)
+	if(!istype(selection))
+		to_chat(user, span_warning("Invalid Image."))
+		return
+	printpicture(user,selection)
+	visible_message(span_notice("A picture appears on top of the chassis of [paimob.name]!"))
```

Comment:
Missing a loc for this visible_message

Reply:
paimob = loc

if this isnt what you mean explain in DIsc.

Reply:
Sorry, this just looked off. The reason is it's missing a "self" version of the message. Check the definition of `proc/visible_message`. You should add that and then remove the `to_chat` call below.

Reply:
should have fixed it!